### ISSUE TITLE
OR operator works with nullable column incorrectly.

### ISSUE THREAD

**Issue by StarWix** [yandex/ClickHouse#1416]
```
CREATE TABLE test
(
    x Nullable(Int32)
) ENGINE = Log

INSERT INTO test VALUES(1), (0), (null)

SELECT * FROM test
┌──x─┐
│  1 │
│  0 │
│ \N │
└────┘

SELECT x FROM test  WHERE x != 0
┌─x─┐
│ 1 │
└───┘

SELECT x FROM test  WHERE x != 0 OR isNull(x)
┌─x─┐
│ 1 │
└───┘

SELECT x FROM test WHERE x != 1
┌─x─┐
│ 0 │
└───┘
```

I expect to get this results
```
// where x != 0
┌──x─┐
│  1 │
│ \N │
└────┘

// where x != 1
┌──x─┐
│  0 │
│ \N │
└────┘
```

**Comment by ludv1x** [comment]
Operator `notEqual` returns Nullable(UInt8) type
```
:) SELECT x, x != 0, isNull(x != 0) FROM test 

SELECT 
    x, 
    x != 0, 
    isNull(x != 0)
FROM test 

┌──x─┬─notEquals(x, 0)─┬─isNull(notEquals(x, 0))─┐
│  1 │               1 │                       0 │
│  0 │               0 │                       0 │
│ \N │              \N │                       1 │
└────┴─────────────────┴─────────────────────────┘
```
and it is compatible with SQL standard.

So, to filter `0` values you should use `x != 0 OR isNull(x)`.
Unfortunately, operator `OR` doesn't work according to SQL standard:
```
:) SELECT x, x != 0 OR isNull(x != 0) FROM test 

SELECT 
    x, 
    (x != 0) OR isNull(x != 0)
FROM test 

┌──x─┬─or(notEquals(x, 0), isNull(notEquals(x, 0)))─┐
│  1 │                                            1 │
│  0 │                                            0 │
│ \N │                                           \N │
└────┴──────────────────────────────────────────────┘
```
(it have to return `1` in the last row).

So, it is better to rename issue like
```
`OR` operator works with nullable column incorrectly
```

**Comment by silviucpp** [comment]
Seems related to https://github.com/yandex/ClickHouse/issues/1626 as well

**Comment by alexey-milovidov** [comment]
Fixed in 04eb0b8
